cradle-core/src/main/java/com/exactpro/cradle/utils/MessageUtils.java

```diff
@@ -46,15 +47,19 @@ public static void validateMessage(CradleMessage message) throws CradleStorageEx
 	{
 		if (message.getId() == null)
 			throw new CradleStorageException("Messages ID cannot be null");
-			
-		if (message.getBookId() == null)
+		if (message.getBookId() == null || StringUtils.isEmpty(message.getBookId().toString()))
 			throw new CradleStorageException("Message must have a book");
-		if (StringUtils.isEmpty(message.getSessionAlias()))
+		if (StringUtils.isEmpty(message.getSessionAlias()) )
 			throw new CradleStorageException("Message must have a session alias");
 		if (message.getDirection() == null)
 			throw new CradleStorageException("Message must have a direction");
 		if (message.getTimestamp() == null)
 			throw new CradleStorageException("Message must have a timestamp");
+		Instant now = Instant.now();
+		if (message.getTimestamp().isAfter(now))
```

Comment:
This check may lead to problem when validation is done not before adding a message. Is it possible?